### DISCUSSION TITLE
Observable's Plot Geo Mark and Projections

### DISCUSSION THREAD

**Discussion by scottfranz** [quarto-dev/quarto-cli#3966] [Q&A] [answered]
Hi,

Just wondering when we will be able to use Plot's newer [Plot.geo mark](https://observablehq.com/@observablehq/plot-geo) and the [Plot projection system](https://observablehq.com/@observablehq/plot-projections)? In addition, Plot just updated their facet option to also work at the mark level. Here is an [example](https://observablehq.com/@observablehq/plot?collection=@observablehq/plot#cell-439).

I tried implementing the geo mark in Quarto and I get a an error. `TypeError: Plot.geo is not a function`

I tried implementing the facet option and it was just ignored.

Please let me know if this would be better if filed as an issue.

**Comment by cscheid** [comment, marked as answer]
We update downstream libraries once per release. So if something is not on 1.3 prerelease, it'll have to wait until 1.4.

In the meantime, you should be able to explicitly import the latest version of Plot by going through skypack.dev.

**Reply by sdwfrost** [reply]
I've tried a few ways to import directly but without success - can you give a one-liner in ojs that will use the latest Plot?

**Reply by cscheid** [reply]
Apparently skypack is now semi-defunct (and it specifically has issues building one of Plot's dependencies). But you can use esm.sh instead:

````
---
title: import Plot library directly
---

```{ojs}
PlotDev = await import("https://esm.sh/@observablehq/plot");
PlotDev
```
````

specifically, note the Geo class there:

<img width="977" alt="image" src="https://user-images.githubusercontent.com/285675/212553261-469eac46-2792-452d-9fe9-95bd1539fef7.png">

**Reply by scottfranz** [reply]
Thanks! This is great.

**Reply by aborruso** [reply]
@cscheid I have created a choropleth map, but when I try to render it (via observable import) in a quarto cell, I have OJS Runtime Error: "Plot.geo is not a function".

Just to see if I understand: right now, I can just import an iframe and wait for the next fourth release?

Thank you

![image](https://user-images.githubusercontent.com/30607/234113271-45cb3758-113b-4a64-9896-607e2a8960eb.png)

**Reply by cscheid** [reply]
If you're importing from a notebook in observablehq.com which requires this new version then yes, you'll need to wait for our dependency refresh in a few weeks.

**Reply by aborruso** [reply]
> If you're importing from a notebook in observablehq.com which requires this new version then yes, you'll need to wait for our dependency refresh in a few weeks.

To use it today in quarto, I must import the updated plot library, and write all the javascript code inside quarto cells. Is it right?

Thank you

**Reply by cscheid** [reply]
This should be coming up in the new 1.4 builds:

<img width="795" alt="image" src="https://user-images.githubusercontent.com/285675/234652114-720353a7-d48a-49e6-af32-2195e7abf9aa.png">

**Reply by cscheid** [reply]
https://github.com/quarto-dev/quarto-cli/pull/5320 this is now merged, so the next 1.4 build will have the new observablehq runtime and stdlib versions.

**Reply by aborruso** [reply]
Wow, really great, thank you very much
